src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -23,34 +23,45 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comments = requestToString(request);
-    float commentsScore = calculateSentimentScore(comments);
+    Float commentsScore = calculateSentimentScore(comments);
     videoAnalysis = new VideoAnalysis(commentsScore);
     String json = new Gson().toJson(videoAnalysis);
 
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-  // Converts JSON String of a request (that is in the form of an array) to a String, separated by
-  // periods. If the Json cannot be converted to an array, an empty String is returned. 
-  // Ex. "["a","b","c"]" --> "a. b. c"
-  public String requestToString(HttpServletRequest request) throws IOException {
+  /**
+   * Converts Json String of a request (that is in the form of an array) to a String, separated by
+   * periods. If the Json cannot be converted to an array or is an empty array, null is returned. 
+   * Ex. "["a","b","c"]" --> "a. b. c"
+   */
+  private String requestToString(HttpServletRequest request) throws IOException {
     Gson gson = new Gson();
     
     String textStr = request.getReader().lines().collect(Collectors.joining());
-    List<String> textList;
+
+    if (textStr.equals("[]")) {
+      return null;
+    }
```

Comment:
How about letting gson convert this into an empty list which then the string join will return an empty string? Seems like the intended behavior of a method called `requestToString`. Then your main method could handle this. I recommend not even moving forward with the rest of the servlet with an empty string and returning an HTTP bad request error saying that the response must be a non-empty array.